src/marshmallow_sqlalchemy/convert.py

```diff
@@ -147,14 +177,17 @@ def property2field(self, prop, *, instance=True, field_class=None, **kwargs):
         if not instance:
             return field_class
         field_kwargs = self._get_field_kwargs_for_property(prop)
-        field_kwargs.update(kwargs)
+        _field_update_kwargs(field_class, field_kwargs, kwargs)
         ret = field_class(**field_kwargs)
         if (
             hasattr(prop, "direction")
             and self.DIRECTION_MAPPING[prop.direction.name]
             and prop.uselist is True
         ):
-            ret = RelatedList(ret, **kwargs)
+            related_list_kwargs = _field_update_kwargs(
+                RelatedList, self.get_base_kwargs(), kwargs
+            )
+            ret = RelatedList(ret, **related_list_kwargs)
```

Comment:
Resolves issue that @AbdealiJK had in #361 where `column='name'` was causing depreciation warning when passed to `RelatedList`.